lib/components/admin/mailables-window.js

```diff
@@ -0,0 +1,194 @@
+      this.setState({mailables})
+    }
+  }
+
+  _removeMailable = (mailable) => {
+    const mailables = [...this.state.mailables]
+    const removeIndex = mailables.findIndex(m => m.name === mailable.name)
+    mailables.splice(removeIndex, 1)
+    this.setState({mailables})
+  }
+
+  _updateMailableField = (index, fieldName, value) => {
+    const mailables = [...this.state.mailables]
+    mailables[index] = {...mailables[index], [fieldName]: value}
+    this.setState({mailables})
+  }
+
+  _onClickCreateLetter = () => createLetter(this.state, this.props.mailablesConfig)
+
+  _updateField = (evt) => {
+    this.setState({[evt.target.id]: evt.target.value})
+  }
+
+  render () {
+    const {mailablesConfig, callTaker, toggleMailables} = this.props
+    const {mailables: selectedMailables} = this.state
+    const {items} = mailablesConfig
+    if (!callTaker.mailables.visible) return null
+    const selectableMailables = items.filter(m =>
+      !selectedMailables.find(mailable => mailable.name === m.name)
+    )
+    return (
+      <DraggableWindow
+        header={
+          <>
+            <WindowHeader>
+              <Icon type='graduation-cap' /> Mailables{' '}
+              <span className='pull-right'>
+                <button
+                  className='clear-button-formatting'
+                  onClick={this._onClickCreateLetter}
+                  style={{marginRight: '5px', verticalAlign: 'bottom'}}
+                >
+                  Create Letter
+                </button>
+              </span>
+            </WindowHeader>
+          </>
+        }
+        onClickClose={toggleMailables}
+        scroll={false}
+        style={{height: '340px', width: '600px'}}
+      >
+        <div>
+          {MAILABLE_FIELDS.map(f => (
+            <input
+              key={f.fieldName}
+              id={f.fieldName}
+              onChange={this._updateField}
+              placeholder={f.placeholder}
+              style={{margin: '5px'}}
+              value={this.state[f.fieldName]} />
+          ))}
+        </div>
+        <div style={{display: 'flex'}}>
+          <div style={{width: '300px'}}>
+            <h4>All Mailables</h4>
+            <MailablesList>
+              {selectableMailables.map((mailable, i) => (
+                <MailableOption
+                  key={mailable.name}
+                  mailable={mailable}
+                  onClick={this._addMailable} />
+              ))}
+            </MailablesList>
+          </div>
+          <div style={{width: '300px'}}>
+            <h4>Selected Mailables <Badge>{selectedMailables.length}</Badge></h4>
+            <MailablesList>
+              {selectedMailables.length > 0
+                ? selectedMailables.map((mailable, i) => (
+                  <MailableOption
+                    index={i}
+                    key={mailable.name}
+                    mailable={mailable}
+                    onClear={this._removeMailable}
```

Comment:
Can this prop be renamed `onRemove` to be consistent with the handler name?

Reply:
Addressed in f04d90a0fbe5cd58d4c9461637031a23787d7a29